File: BE/src/main/java/orion/rs/demo/domain/Transaction.java
package orion.rs.demo.domain;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Transaction {

    @Id
    @GeneratedValue
    private Long id;

    @ManyToOne
    private Employee reporter;

    @ManyToOne
    private Account account;

    private String description;

    @Version
    private Integer version;

    private Date date;

    private BigDecimal amount;

    private String category;
    @Enumerated(EnumType.STRING)
    @NotNull
    private Status status;

}

File: FE/app/src/main/java/com/orioninc/financetracker/model/Transaction.kt
package com.orioninc.financetracker.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.util.Date

enum class Status { Pending, Completed, Reconciled}

data class Transaction(
    @SerializedName("id")
    val idTransaction: Long,
    @SerializedName("reporter")
    val reporter: Employee,
    @SerializedName("account")
    val account: Account,
    @SerializedName("description")
    val description: String,
    @SerializedName("version")
    val version: Int,
    @SerializedName("date")
    val date: Date,
    @SerializedName("amount")
    val amount: Double,
    @SerializedName("category")
    val category: String,
    @SerializedName("status")
    val status: Status
): Serializable